### DISCUSSION TITLE
Bathymetry & Sill Depth Modifications

### DISCUSSION THREAD

**Discussion by gknorr** [esm-tools/esm_tools#619] [Models]
Hello all,

a potential workflow for Bathymetry and Sill depth modifications is provided here; please see the attached [PDF file](https://github.com/esm-tools/esm_tools/files/8316143/Bathymetry.chnages.pdf).

If any particular problems arise, please document them here so we can maintain a running record of problems that may arise. Note please that any serious problems may be moved to a separate thread to be more discoverable.

Happy modelling,
Paul & Gregor

**Comment by fernandadialzira** [comment]
Thank you @pgierz and @gknorr for working on this method! I would like to lay out what I am trying to do here, in order to check some steps of the tutorial. The idea is that we have a paleoMIST mesh for 37.5ka, but Bering Strait and Panama are open, so I need to close them in the mesh. I followed the beginning of the instructions using Paul's Notebook: `/work/ollie/fmatos/paul_notebooks/mesh_adjust_bathymetry.ipynb` and identified 5 nodes on the Panama region to start testing. Then I followed Gregor's instructions and did:

```
awk 'NR==60639 {$0="0.000"} NR==60640 {$0="0.000"} NR==60641 {$0="0.000"} NR==60642 {$0="0.000"} NR==60643 {$0="0.000"}{print}' aux3d.out > aux3d_new38ka.out
```
Is this correct for my purpose?
On gregor's pdf there are further steps to use the new mesh for simulations, but right now I would just like to visualize the result and see whether it worked out. However, I tried renaming the new `aux3d_new38ka.out` to `aux3d.out` and visualizing it with the notebook and no change was there. Is it needed for visualization that I start step 3 (Recreate the distance info in the mesh folder dist_288)? Or is there another step in between to visualize the change done to this file?

The mesh is at `/work/ollie/fmatos/mis3/boundary_conditions/paleomist37.5ka/fesom/mesh_CORE2_GLAC1D37.5`

**Comment by gknorr** [comment]
The method described does not allow you to introduce e.g. new land points. But if you change the depth entries at existing nodes, the changes should be visible directly. The recalculation of the distances (Step 3) is necessary only before you run the model.

**Comment by pgierz** [comment]
Hi Fernanda, for closing off new land points, the raw input data must be modified with, e.g. `cdo`.

For instance:
```
$ cdo setclonlatbox,0,lon0,lon1,lat0,lat1 <in> <out>
``` 

I might be misremembering the argument order, but in any case this will help:
```
$ cdo -h setclonlatbox
```

There is alternatively also the `indexbox` equivalent which takes directly the array indices rather than coordinates.

**Comment by pgierz** [comment]
For future reference:
```
cdo -h setclonlatbox
NAME
    setclonlatbox, setcindexbox - Set a box to constant

SYNOPSIS
    setclonlatbox,c,lon1,lon2,lat1,lat2  infile outfile
    setcindexbox,c,idx1,idx2,idy1,idy2  infile outfile

DESCRIPTION
    Sets a box of the rectangularly understood field to a constant value. The elements outside 
    the box are untouched, the elements inside are set to the given constant. All input fields 
    need to have the same horizontal grid.

OPERATORS
    setclonlatbox  Set a longitude/latitude box to constant
                   Sets the values of a longitude/latitude box to a constant value. The 
                   user has to give the longitudes and latitudes of the edges of the box.
    setcindexbox   Set an index box to constant
                   Sets the values of an index box to a constant value. The user has to 
                   give the indices of the edges of the box. The index of the left edge 
                   can be greater than the one of the right edge.

PARAMETER
    c     FLOAT    Constant
    lon1  FLOAT    Western longitude
    lon2  FLOAT    Eastern longitude
    lat1  FLOAT    Southern or northern latitude
    lat2  FLOAT    Northern or southern latitude
    idx1  INTEGER  Index of first longitude
    idx2  INTEGER  Index of last longitude
    idy1  INTEGER  Index of first latitude
    idy2  INTEGER  Index of last latitude
```
